iOS_SDK/OneSignalSDK/Source/OSInAppMessageViewController.m

```diff
@@ -392,47 +395,58 @@ Dismisses the message view with a given direction (up or down) and velocity
  */
 - (void)dismissCurrentInAppMessage:(BOOL)up withVelocity:(double)velocity {
     
-    // Inactivate the current Y constraints
-    self.finalYConstraint.active = false;
-    self.initialYConstraint.active = false;
-    
-    // The distance that the dismissal animation will travel
-    var distance = 0.0f;
-    
-    // Add new Y constraints
-    if (up) {
-        distance = self.messageView.frame.origin.y + self.messageView.frame.size.height + 8.0f;
-        [self.messageView.bottomAnchor constraintEqualToAnchor:self.view.topAnchor constant:-8.0f].active = true;
-    } else {
-        distance = self.view.frame.size.height - self.messageView.frame.origin.y + 8.0f;
-        [self.messageView.topAnchor constraintEqualToAnchor:self.view.bottomAnchor constant:8.0f].active = true;
-    }
-    
-    var dismissAnimationDuration = velocity != 0.0f ? distance / fabs(velocity) : 0.3f;
-    
-    var animationOption = UIViewAnimationOptionCurveLinear;
-    
-    // Impose a minimum animation speed (max duration)
-    if (dismissAnimationDuration > MAX_DISMISSAL_ANIMATION_DURATION) {
-        animationOption = UIViewAnimationOptionCurveEaseIn;
-        dismissAnimationDuration = MAX_DISMISSAL_ANIMATION_DURATION;
-    } else if (dismissAnimationDuration < MIN_DISMISSAL_ANIMATION_DURATION) {
-        animationOption = UIViewAnimationOptionCurveEaseIn;
-        dismissAnimationDuration = MIN_DISMISSAL_ANIMATION_DURATION;
-    }
+    // Validate whether or not WebView actually rendered the IAM code, which we lead to global constraits not being nil
+    if (self.didPageRenderingComplete) {
```

Comment:
Lets clean this large if block by checking for `!self.didPageRenderingComplete` first and returning ealy, since this code section is much smaller.